### PR TITLE
Add tests for empty or missing nonce

### PR DESCRIPTION
**Pull request by mida-digg**
Add tests for empty or missing nonce when creating a WUA.

## Checklist

- [x] Changes are limited to a single goal (avoid scope creep)
- [x] I confirm that I have read any Contribution and Development guidelines (CONTRIBUTING and DEVELOPMENT) and are following their suggestions.
- [x] I confirm that I wrote and/or have the right to submit the contents of my Pull Request, by agreeing to the _Developer Certificate of Origin_, (adding a 'sign-off' to my commits).

### CODE REVIEW

File: src/test/java/se/digg/wallet/ecosystem/WalletProviderTest.java [hunk 1]
```diff
@@ -42,4 +43,24 @@ void createsWalletUnitAttestationV2() throws Exception {
     assertThat(wua, matchesPattern(
         "^[A-Za-z0-9]+\\.[A-Za-z0-9]+\\.[A-Za-z0-9\\-_]+$"));
   }
+
+  @Test
+  void createsWalletUnitAttestationV2_withEmptyNonce() throws Exception {
```
Comment (eric-thelin) [inline src/test/java/se/digg/wallet/ecosystem/WalletProviderTest.java]: Jag tror att vi kan baka ihop de tre testerna `createsWalletUnitAttestationV2`, `createsWalletUnitAttestationV2_withEmptyNonce`, `createsWalletUnitAttestationV2_withNullNonce` till ett enda parameteriserat test:
```
@ParameterizedTest
@ValueSource(strings = {"nonce", ""})
@NullSource
void createsWalletUnitAttestationV2(String nonce) throws Exception {
  String wua = walletProvider.getWalletUnitAttestationV2(
      new ECKeyGenerator(Curve.P_256).generate(),
      nonce);

  assertThat(wua, matchesPattern(
      "^[A-Za-z0-9]+\\.[A-Za-z0-9]+\\.[A-Za-z0-9\\-_]+$"));
}
```

**Review by eric-thelin — COMMENTED** [review]
Jag kommenterade kring en möjlig förbättring för att reducera mängden duplicerad testkod genom att använda JUnits stöd för parameteriserade tester.

Jag tror också att det vore bra att i WalletClientGatewayTest lägga till ett test som skickar nonce till /wua/v3. I dagsläget ser det ut som att vi använder /wua/v3 utan nonce. Det går antagligen också att göra som ett parametriserat test för att testa båda fallen.

**Review by eric-thelin — APPROVED** [review]
